documentation/working-with-openfire.html

```diff
@@ -1,47 +1,97 @@
-                    <p>From a terminal in the root of Openfire run:</p>
-                    <p class="code">
-                        ./build/docker/buildWithDocker.sh
-                        docker build . -t openfire:mything
-                    </p>
-        
-                    <h4>Running an Openfire container image</h4>
-                    <p>There are quite a lot of ports involved, or there can be, depending on what you're doing. This list includes pretty much everything. Some of them may conflict with things already running, in which case you'll either need remap, or switch port and reconfigure Openfire accordingly</p>
-                    <p class="code">
-                        docker run --rm -it -p 3478:3478/tcp -p 3479:3479/tcp -p 5222:5222/tcp -p 5223:5223/tcp -p 5229:5229/tcp -p 5262:5262/tcp -p 5263:5263/tcp -p 5275:5275/tcp -p 5276:5276/tcp -p 7070:7070/tcp -p 7443:7443/tcp -p 7777:7777/tcp -p 9090:9090/tcp -p 9091:9091/tcp -p 5005:5005/tcp openfire:mything
-                    </p>
-        
-                    <p>In fact, I've used this so often, I have it as a Bash alias</p>
-                    <pre class="code">
-function ofdocker(){
+
+<article>
+
+    <header>
+        <img src="images/header_logo.gif" alt="Openfire Logo" />
+        <h1>Working with Openfire</h1>
+    </header>
+
+    <nav>
+        <a href="index.html">&laquo; Back to documentation index</a>
+    </nav>
+
+    <section id="intro">
+
+        <h2>Introduction</h2>
+        <p>Collected tips &amp; tricks for developers and other software engineers working with Openfire</p>
+        <p><a href="https://github.com/igniterealtime/Openfire/edit/main/documentation/working-with-openfire.html">Add your own on GitHub!</a></p>
+
+        <p>
+            The following sections are available in this document:
+        </p>
+        <nav>
+            <ul>
+                <li><a href="#">Tooling</a>
+                    <ul>
+                        <li><a href="#tooling-docker">Docker</a></li>
+                        <li><a href="#tooling-spark">Spark Debug Tools</a></li>
+                        <li><a href="#tooling-sint">Smack Integration Tests</a></li>
+                        <li><a href="#tooling-dependencies">Dependencies</a></li>
+                        <li><a href="#tooling-dockerstacks">Docker Stacks</a></li>
+                        <li><a href="#tooling-plugins">Useful Plugins</a></li>
+                    </ul>
+                </li>
+                <li><a href="#links">Useful Links</a>
+                    <ul>
+                        <li><a href="#links-github">GitHub</a></li>
+                        <li><a href="#links-standards">Standards</a></li>
+                        <li><a href="#links-chat">Chat</a></li>
+                    </ul>
+                </li>
+            </ul>
+        </nav>
+
+    </section>
+
+    <section>
+
+        <h2>Tooling</h2>
+
+        <section id="tooling-docker">
+
+            <h3>Docker</h3>
+
```

Comment:
Should the docker tips be merged with / linked to the main docker page?
It's out of scope for this - another ticket to be made?

Reply:
I have no strong opinion either way. If we do go for that, lets indeed take it on under a different ticket.

Reply:
https://igniterealtime.atlassian.net/browse/OF-2554